optaplanner-core/src/main/java/org/optaplanner/core/impl/score/buildin/bendable/BendableScoreDefinition.java

```diff
@@ -109,7 +110,7 @@ public BendableScoreInliner buildScoreInliner(boolean constraintMatchEnabled) {
     }
 
     @Override
-    public BendableScoreHolder buildScoreHolder(boolean constraintMatchEnabled) {
+    public ScoreHolder buildScoreHolder(boolean constraintMatchEnabled) {
```

Comment:
I am not sure about these changes.
In 8.0, where we have `DefaultBendableScoreHolder implements BendableScoreHolder`, we'll want to return `BendableScoreHolder` here (not `DefaultBendableScoreHolder` nor just `ScoreHolder`). Or wait - optaplanner-core will need something that has extractScore(), so it will need to return AbstractScoreHolder or InnerScoreHolder or DefaultBendableScoreHolder anyway...

Reply:
ok to change like this or to leave as is. your call - the real work with be in the 8.x branch